telemetry/telemetry.ts

```diff
@@ -1,35 +1,15 @@
-import vscode = require('vscode');
-import TelemetryReporter from 'vscode-extension-telemetry';
-
-export let reporter: TelemetryReporter;
-
-export class Reporter extends vscode.Disposable {
-
-    constructor(ctx: vscode.ExtensionContext) {
-
-        super(() => reporter.dispose());
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
 
-        let packageInfo = getPackageInfo(ctx);
-        reporter = packageInfo && new TelemetryReporter(packageInfo.name, packageInfo.version, packageInfo.aiKey);
-
-    }
-}
+import vscode = require('vscode');
+import { createTelemetryReporter, ITelemetryReporter } from 'vscode-azureextensionui';
+import { ext } from '../extensionVariables';
 
-interface IPackageInfo {
-    name: string;
-    version: string;
-    aiKey: string;
-}
+export let reporter: ITelemetryReporter;
 
-function getPackageInfo(context: vscode.ExtensionContext): IPackageInfo {
-    // tslint:disable-next-line:non-literal-require
-    let extensionPackage = require(context.asAbsolutePath('./package.json'));
-    if (extensionPackage) {
-        return {
-            name: extensionPackage.name,
-            version: extensionPackage.version,
-            aiKey: extensionPackage.aiKey
-        };
-    }
-    return;
+export function initializeTelemetryReporter(ctx: vscode.ExtensionContext): void {
+    reporter = createTelemetryReporter(ctx);
```

Comment:
Can we just get rid of this file now and move these two lines over to dockerExtension.ts?

Reply:
Almost. Lots of places use the reporter from this file, I'll be getting rid of it once I've moved to using the shared registerCommand.